js_modules/dagit/packages/core/src/assets/AssetsCatalogTable.tsx

```diff
@@ -43,7 +46,54 @@ export const AssetsCatalogTable: React.FC<{prefixPath?: string[]}> = ({prefixPat
     skip: view === 'graph',
   });
 
-  const refreshState = useQueryRefreshAtInterval(assetsQuery, FIFTEEN_SECONDS);
+  const assetsOrError = assetsQuery.data?.assetsOrError;
+  const assets = assetsOrError?.__typename === 'AssetConnection' ? assetsOrError.nodes : undefined;
+
+  const searchSeparatorAgnostic = (search || '')
+    .replace(/(( ?> ?)|\.|\/)/g, '>')
+    .toLowerCase()
+    .trim();
+
+  const filtered = React.useMemo(
+    () =>
+      (assets || []).filter(
+        (a) =>
+          !searchSeparatorAgnostic ||
+          tokenForAssetKey(a.key).toLowerCase().includes(searchSeparatorAgnostic),
+      ),
+    [assets, searchSeparatorAgnostic],
+  );
+
+  const {displayPathForAsset, displayed, nextCursor, prevCursor} =
+    view === 'flat'
+      ? buildFlatProps(filtered, prefixPath, cursor)
+      : buildNamespaceProps(filtered, prefixPath, cursor);
+
+  const displayedKeys = React.useMemo(
+    () => displayed.map<AssetKey>((a) => ({path: a.key.path})),
+    [displayed],
+  );
+  const displayedDefinitionMap = React.useMemo(
+    () =>
+      Object.fromEntries(
+        displayed
+          .filter((n) => n.definition)
+          .map((n) => [toGraphId(n.key), {definition: n.definition!}]),
+      ),
+    [displayed],
+  );
+
+  const {liveDataByNode, liveResult} = useLiveDataForAssetKeys(
+    null,
+    displayedDefinitionMap,
+    null,
+    displayedKeys,
```

Comment:
Yeah I agree with this - I ended up breaking the "compute status" part out of this hook so that there are fewer arguments and I think its' ok now. (Hopefully soon the compute status stuff will move to the backend)